pkg/controllers/dynakube/controller.go

```diff
@@ -100,13 +103,19 @@ func (controller *Controller) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(controller)
 }
 
+type dtclientReconciler interface {
```

Comment:
the name reads like that it will reconcile the `dtclient` and not use it

having it in this package feels a bit odd aswell

what if there was a `reconciler` package, that had an the `interface`(s) in it, and call it something like `WithDtClient`?
```go
k8sEntityReconciler reconciler.WithDtClient
```
`WithDtClient` is a bad name, could be `DTReader` but that would imply that it doesn't "write/post", which would be true in your case, but the `dtclient` makes no such difference 😅 

could go very simple `reconciler.DT`, implying that this is a reconciler that "reconcilers according to the Dynatrace Environment"

Reply:
ok, we can extract it later if it make sense (if we want to reuse the interface anywhere else)